### PR TITLE
Typegql

### PR DESCRIPTION
**Pull request by modelB**
fixes to schema/resolvers to accommodate datatype conversion between postgres and graphql better, and work around regex

### CODE REVIEW

File: server/gqlcorp/typeMain.js [hunk 1]
```diff
@@ -135,7 +135,8 @@ const typeCreator = (nonJoinTables, fktObj, fktObjNoJoins, nullable) => {
           // but if it's in the foreign key object, it adds the type as a connection to a different table.
           const temp = fktObj[key][prop]
           // const temp = camelCase(fktObj[key][prop]);
-          typeObj[key][temp] = '['+capFirstLet(singular(temp))+']';
+          // typeObj[key][temp] = '['+capFirstLet(singular(temp))+']';
+          typeObj[key][temp] = '['+temp.slice(0,1).toUpperCase()+camelCase(singular(temp)).slice(1)+']';
```
Comment (AdamrG1) [inline server/gqlcorp/typeMain.js:139]: What is the purpose of slice here?

File: server/gqlcorp/mutation.js [hunk 2]
```diff
@@ -31,6 +31,16 @@ const convertTypesforMutation = object => {
...
           object[table][column] = "String";
           break;
+        case 'numeric':
+          object[table][column] = 'Int'
+        break;
+        case 'smallint':
+          object[table][column] = 'Int'
+          break;
+        case 'ARRAY':
+          object[table][column] = '|String|'
```
Comment (markyliu) [inline server/gqlcorp/mutation.js:41]: not sure if specific enough to handle all array cases in long run; they are custom in gql

**Review by AdamrG1 — APPROVED** [review]
LGTM

File: server/gqlcorp/mutation.js (hunk 2 above)
Comment (markyliu) [inline server/gqlcorp/mutation.js:41]: SQL to gql? custom type in gql necessary

File: server/gqlcorp/typeMain.js (hunk 1 above)
Comment (markyliu) [reply server/gqlcorp/typeMain.js:139]: should be pascal function use, resolve later

File: server/gqlcorp/typeMain.js [hunk 3]
```diff
@@ -149,7 +150,7 @@ const typeCreator = (nonJoinTables, fktObj, fktObjNoJoins, nullable) => {
       if (arrFK.includes(table)) {
         const valsToInput = arrFK.filter(elem => elem !== table) //filter for all other tables
         valsToInput.forEach(val => { //Add to final table 
-          typeObj[table][val] = '['+capFirstLet(singular(val))+']'
+          typeObj[table][val] = '['+val.slice(0,1).toUpperCase()+camelCase(singular(val)).slice(1)+']'
```
Comment (markyliu) [inline server/gqlcorp/typeMain.js:153]: use written function

File: server/gqlcorp/typeMain.js [hunk 4]
```diff
@@ -182,7 +183,7 @@ for (const njtCol in nonJoinTables) { //type object name i.e. planets, species,
           if (!(Object.keys(typeObj[njtCol]).includes(fktCol))) {
             //console.log('missing fks', typeObj[njtCol],fktCol)
             let temp = camelCaseIt(fktCol);
-            typeObj[njtCol][temp] = '['+snakeToTitle(singular(fktCol))+']'
+            typeObj[njtCol][temp] = '['+fktCol.slice(0,1).toUpperCase()+camelCase(singular(fktCol)).slice(1)+']'
```
Comment (markyliu) [inline server/gqlcorp/typeMain.js:186]: use written function

File: server/gqlcorp/typeMain.js [hunk 5]
```diff
@@ -194,8 +195,12 @@ for (const njtCol in nonJoinTables) { //type object name i.e. planets, species,
 
   //convert SQL types to GQL type
   for (const table in typeObj) {
-
+    
     for (const column in typeObj[table]) {
+      if (nonJoinTables.hasOwnProperty(column) && camelCase(column) !== column) {
+        typeObj[table][camelCase(column)] = typeObj[table][column];
+        delete typeObj[table][column];
```
Comment (markyliu) [inline server/gqlcorp/typeMain.js:202]: mutation of object..?

**Review by markyliu — APPROVED** [review]
things to go back to after launch but lgtm

**Pull request merged**